specification/gsm/resource-manager/Microsoft.ManagementPartner/preview/2018-02-01/gsm.json

```diff
@@ -0,0 +1,428 @@
+      "put": {
+        "summary": "Create a specific `Partner`.",
+        "description": "Create a management partner for the objectId and tenantId.",
+        "operationId": "Partner_Create",
+        "x-ms-examples": {
+          "PutPartnerDetails": {
+            "$ref": "./examples/PutPartnerDetails.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/PartnerIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "tags": [
+          "partner"
+        ],
+        "responses": {
+          "200": {
+            "description": "Get the details of the `Partner`.",
+            "schema": {
+              "$ref": "#/definitions/PartnerResponse"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "summary": "Update a specific `Partner`.",
+        "description": "Update the management partner for the objectId and tenantId.",
+        "operationId": "Partner_Update",
+        "x-ms-examples": {
+          "PatchPartnerDetails": {
+            "$ref": "./examples/PatchPartnerDetails.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/PartnerIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "tags": [
+          "partner"
+        ],
+        "responses": {
+          "200": {
+            "description": "Get the details of the `Partner`.",
+            "schema": {
+              "$ref": "#/definitions/PartnerResponse"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "delete": {
+        "summary": "Delete a specific `Partner`.",
+        "description": "Delete the management partner for the objectId and tenantId.",
+        "operationId": "Partner_Delete",
+        "x-ms-examples": {
+          "DeletePartnerDetails": {
+            "$ref": "./examples/DeletePartnerDetails.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/PartnerIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "tags": [
+          "partner"
+        ],
+        "responses": {
+          "200": {
+            "description": "Deleted the `Partner`."
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.ManagementPartner/partners": {
+      "get": {
```

Comment:
Why are there 2 gets for partner one with and without the id? In the case of the one without the partner Id why is this not a list?

Reply:
@simongdavies  this is a good question. Because each user under one tenant can only have 1 partner id, that's why it doesn't return a list. 
Technically, the call doesn't need to pass the partner id, but if the call passes the partner id, we will compare if it is match our record, it doesn't match, the call will fail.

Reply:
@simongdavies you are right, we only need one get operation, the partner is optional. Delete the one without partner id. Can you please approve